backend/api/Controllers/RobotController.cs

```diff
@@ -654,6 +657,99 @@ public async Task<ActionResult> ResumeMission([FromRoute] string robotId)
         return Ok(mission);
     }
 
+
+    /// <summary>
+    /// Starts a mission which drives the robot to the nearest safe position
+    /// </summary>
+    /// <remarks>
+    /// <para> This query starts a localization for a given robot </para>
+    /// </remarks>
+    [HttpPost]
+    [Route("{robotId}/{asset}/{deck}/safe-position")]
+    [Authorize(Roles = Role.User)]
+    [ProducesResponseType(typeof(Mission), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<Mission>> SafePosition(
+        [FromRoute] string robotId,
+        [FromRoute] string asset,
+        [FromRoute] string deck
+    )
+    {
+        var robot = await _robotService.ReadById(robotId);
+        if (robot == null)
+        {
+            _logger.LogWarning("Could not find robot with id={id}", robotId);
+            return NotFound("Robot not found");
+        }
+
+        var assetDeck = await _assetDeckService.ReadByAssetAndDeck(asset, deck);
+        if (assetDeck is null)
+        {
+            _logger.LogWarning("Could not find any safe positions for asset={asset} and deck={deck}", asset, deck);
+            return NotFound("No safe position found");
```

Comment:
I think this should be split in 3 checks, one for asset, then for deck, then for safe positions

Reply:
Do you think this is necessary? Same message and status code for all the checks, just more code for the same outcome?

Reply:
Not the same message, "Asset not found", "Deck not found" and "No safe positions on this asset deck"